ssl/record/rec_layer_s3.c

```diff
@@ -913,10 +947,21 @@ int ssl3_write_pending(SSL *s, int type, const unsigned char *buf,
         clear_sys_error();
         if (s->wbio != NULL) {
             s->rwstate = SSL_WRITING;
+            if (BIO_get_offload_tx(s->wbio) && type !=
+                    SSL3_RT_APPLICATION_DATA) {
+                BIO_set_offload_tx_ctrl_msg(s->wbio, type);
+            }
             i = BIO_write(s->wbio, (char *)
                           &(SSL3_BUFFER_get_buf(&wb[currbuf])
                             [SSL3_BUFFER_get_offset(&wb[currbuf])]),
                           (unsigned int)SSL3_BUFFER_get_left(&wb[currbuf]));
+            /* To prevent coalescing of control and data messages,
+             * such as in buffer_write, we flush the BIO
+             */
+            if (BIO_get_offload_tx(s->wbio) &&
+                    type != SSL3_RT_APPLICATION_DATA) {
+                (void)BIO_flush(s->wbio);
```

Comment:
So does that mean if multiple reneg handshake messages are all sent one after the other they are all going to end up in separate TCP packets rather than a single one?